### PR TITLE
handles NPE

### PR DESCRIPTION
**Pull request by jackstine**
# Description

There is a NPE that crashes gloo in the `MarkPerFilterConfig()`.  This will prevent the NPE. 

# Context

Users are facing an issue when the gloo pod crashes because the destination is set to `nil`. Context as to why the destination is set to `nil` is not known yet.
# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/pkg/plugins/pluginutils/per_filter_config.go [hunk 1]
```diff
@@ -118,7 +118,11 @@ func ModifyPerFilterConfig(
 		return nil
 	}
 
-	err = errors.Errorf("unexpected destination type %v", reflect.TypeOf(inAction.GetDestination()).Name())
+	err = errors.New("unexpected destination type that is nil")
```
Comment (sam-heilbron) [inline projects/gloo/pkg/plugins/pluginutils/per_filter_config.go:121]: looks good to me!
could this also be handled by adding a default case to the above switch statement?

Comment (jackstine) [reply projects/gloo/pkg/plugins/pluginutils/per_filter_config.go:121]: Why does that matter

Comment (sam-heilbron) [reply projects/gloo/pkg/plugins/pluginutils/per_filter_config.go]: Oh sorry wasn't proposing that we do it that way, was more just clarifying my understanding of the change

**Review by jenshu — APPROVED** [review]

**Pull request merged**